main.c

```diff
@@ -404,6 +405,11 @@ char* decode_domain_name(const uint8_t** buffer)
     int len = buf[i];
     i += 1;
 
+    if ((j + len) >= 256) {
+      // OVERFLOW ISSUE
+      return OVERFLOW_QNAME;
```

Comment:
Can't you just return NULL and then check for that?

Reply:
Btw., `decode_domain_name` is probably also buggy in other aspect and needs to be rewritten.